### PR TITLE
[RFR] cleaned up hound errors

### PR DESCRIPTION
**Pull request by jacksy40**

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -15,37 +15,78 @@
 //= require foundation
 //= require_tree .
 
-$(function(){ $(document).foundation(); });
```
Comment (radavis) [inline app/assets/javascripts/application.js]: You probably want to keep this line. It enables the Foundation javascript libraries.

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -15,37 +15,78 @@
...
+    data: { body: body, votes: plusVotes, vote_type: voteType }
+  });
+  votesRequest.done(function() {
+    $('.votes-' + number).text(plusVotes);
+  });
+  votesRequest.fail(function() {
+    alert('You\'ve already voted');
   });
 });
+
```
Comment (radavis) [inline app/assets/javascripts/application.js]: Your application.js file is going to get quite large if all your javascript just gets dumped into this file. I would extract the javascript associated with the voting feature into a file called `app/javascripts/votes.js` and add a `//= require votes` right before the `//= require_tree .` statement.

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -18,12 +18,32 @@ def update
...
-      @review.errors.full_messages.each { |message| flash[:errors] = message }
-      render 'movies/show'
+      if @review.update(review_params)
+        redirect_to movie_path(@movie)
+      else
+        @review.errors.full_messages.each { |message| flash[:errors] = message }
+        render 'movies/show'
+      end
     end
   end
```
Comment (radavis) [inline app/controllers/reviews_controller.rb]: You should have some test coverage here. Either a [controller test](https://www.relishapp.com/rspec/rspec-rails/docs/controller-specs) or a [capybara test with a javascript driver](https://robots.thoughtbot.com/how-we-test-rails-applications#YweRys-indicator)

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (jacksy40) [inline app/assets/javascripts/application.js]: already in application.rb

File: app/assets/javascripts/application.js [hunk 4]
```diff
@@ -1,51 +1,71 @@
...
+  var reviewId = $(this).attr('review_id');
+  var voteType = $(this).parent().attr('class');
+  var body = $('.review-body-' + number).text();
+  var votes = $('.votes-' + number).text();
+  var plusVotes = parseInt(votes) - 1;
+
+  var votesRequest = $.ajax({
+    method: 'PUT',
+    url: '/movies/' + movieId + '/reviews/' + reviewId,
+    data: { body: body, votes: plusVotes, vote_type: voteType }
```
Comment (jacksy40) [inline app/assets/javascripts/application.js]: vote_type in this case is the column name

File: spec/features/votes/user_votes_spec.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+} do
+  let(:user) { FactoryGirl.create(:user) }
+
+  before :each do
+    sign_in_as user
+  end
+
+  scenario 'upvote', js: true do
+    movie = FactoryGirl.create(:movie)
+    review = FactoryGirl.create(:review, movie: movie, user: user)
```
Comment (jacksy40) [inline spec/features/votes/user_votes_spec.rb]: changed

File: spec/features/votes/user_votes_spec.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    find('.vote-down').click_link 'Down vote'
+
+    within 'p.votes' do
+      expect(page).to have_content('-1')
+    end
+  end
+
+  scenario 'upvote', js: true do
+    movie = FactoryGirl.create(:movie)
+    review = FactoryGirl.create(:review, movie: movie, user: user)
```
Comment (jacksy40) [inline spec/features/votes/user_votes_spec.rb]: changed

File: spec/features/votes/user_votes_spec.rb [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  before :each do
+    sign_in_as user
+  end
+
+  scenario 'upvote', js: true do
+    movie = FactoryGirl.create(:movie)
+    review = FactoryGirl.create(:review, movie: movie, user: user)
+
+    visit movie_path(movie)
+    #save_screenshot("capybara-js.png")
```
Comment (jacksy40) [inline spec/features/votes/user_votes_spec.rb]: deleted

File: spec/features/votes/user_votes_spec.rb [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+      expect(page).to have_content('1')
+    end
+  end
+
+  scenario 'downvote', js: true do
+    movie = FactoryGirl.create(:movie)
+    review = FactoryGirl.create(:review, movie: movie, user: user)
+
+    visit movie_path(movie)
+    #save_screenshot("capybara-js.png")
```
Comment (jacksy40) [inline spec/features/votes/user_votes_spec.rb]: deleted

File: spec/features/votes/user_votes_spec.rb [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+      expect(page).to have_content('-1')
+    end
+  end
+
+  scenario 'upvote', js: true do
+    movie = FactoryGirl.create(:movie)
+    review = FactoryGirl.create(:review, movie: movie, user: user)
+
+    visit movie_path(movie)
+    #save_screenshot("capybara-js.png")
```
Comment (jacksy40) [inline spec/features/votes/user_votes_spec.rb]: deleted

File: spec/support/database_cleaner.rb [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+  config.before(:suite) do
+    DatabaseCleaner.clean_with(:truncation)
+  end
+
+  config.before(:each) do
+    DatabaseCleaner.strategy = :transaction
+  end
+
+  config.before(:each, :js => true) do
```
Comment (jacksy40) [inline spec/support/database_cleaner.rb]: changed to js: :true

**Pull request closed without merging**